pkg/cni/conf/confmanager.go

```diff
@@ -120,6 +136,34 @@ func (mgr *Manager) insertCNIPluginConf() error {
 	return nil
 }
 
+func (mgr *Manager) removeCNIPluginConf() error {
+	log := logger.GetLogger()
+	cm := &corev1.ConfigMap{}
+	cmKey := client.ObjectKey{Name: mgr.cniUninstallConfigMapName, Namespace: os.Getenv(consts.PodNamespaceEnvKey)}
+	err := mgr.k8sClient.Get(context.Background(), cmKey, cm)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get cni uninstall configMap (%s/%s): %w", cmKey.Namespace, cmKey.Name, err)
+	} else if err == nil {
+		if cm.Data["uninstall"] == "false" {
+			log.Info("Uninstall flag is NOT set, skip removing cni configuration file")
+			return nil
+		}
+	}
+
+	log.Info("Removing cni configuration file...")
+	file := filepath.Join(mgr.cniConfDir, mgr.cniConfFile)
+	if _, err := os.Stat(file); err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to stat file %s: %w", file, err)
+	}
+	if err := os.Remove(file); err != nil {
```

Comment:
add retry here since this conf file might be hold by others. And also set the pod graceful terminating period if your retry interval is longer than default terminating time.

Reply:
Updated both retry logic and podTerminationPeriod. Thanks